File: app/src/main/java/net/emojiparty/android/jishotomo/data/models/SenseWithCrossReferences.java
package net.emojiparty.android.jishotomo.data.models;

import androidx.room.Embedded;
import androidx.room.Ignore;
import java.util.ArrayList;
import java.util.List;
import net.emojiparty.android.jishotomo.data.room.Sense;

public class SenseWithCrossReferences {
  @Embedded public Sense sense;

  @Ignore
  public List<CrossReferencedEntry> crossReferences = new ArrayList<>();

  public Sense getSense() {
    return sense;
  }

  public List<CrossReferencedEntry> getCrossReferences() {
    return crossReferences;
  }

  public void setCrossReferences(List<CrossReferencedEntry> crossReferences) {
    this.crossReferences = crossReferences;
  }
}

File: app/src/main/java/net/emojiparty/android/jishotomo/data/models/SenseWithCrossReferences.kt
package net.emojiparty.android.jishotomo.data.models

import androidx.room.Embedded
import androidx.room.Ignore
import net.emojiparty.android.jishotomo.data.room.Sense

data class SenseWithCrossReferences(
  @Embedded var sense: Sense
) {
  // TODO do i need this class??
  @Ignore
  var crossReferences: List<CrossReferencedEntry> = emptyList()
}
